src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java

```diff
@@ -383,18 +389,31 @@ protected void gotoTargetFloor(ActionEvent event) {
             log(Messages.getString("invalidSelection"));
     }
 
-    private void translateElevator(Integer percentage) {
+    private void translateElevator(Integer percentage, Boolean triggerResize) {
+    	if(triggerResize) {
+    		percentage = oldPercentage;
+    	} else {
+    		oldPercentage = percentage;
+    	}
         double maxHeight = gElevator.getHeight();
         double elevatorHeight = groupElevator.getBoundsInLocal().getHeight();
         double yRect = (maxHeight - elevatorHeight) * percentage / 100.0;
 
         groupElevator.translateYProperty().set(-yRect);
-    }
+        
+    }   
 
     private static class FloorState {
         private final BooleanProperty requestUp = new SimpleBooleanProperty(false);
         private final BooleanProperty requestDown = new SimpleBooleanProperty(false);
         private final BooleanProperty stopRequest = new SimpleBooleanProperty(false);
         private final BooleanProperty isServiced = new SimpleBooleanProperty(false);
     }
+    
+    public void setStageAndSetUpListeners(Stage mainStage) {
+    	stage = mainStage;
```

Comment:
Do we need the stage as a member?

Reply:
Checked it is not necessary as tests showed.